pandora-client-web/src/observable.ts

```diff
@@ -51,6 +50,56 @@ export class Observable<T> implements ReadonlyObservable<T> {
 	}
 }
 
+class RateLimitedObservable<T> extends NormalObservable<T> {
+	private static timeWindow = 1000;
+	private static maxEvents = 10;
+	private static maxDelay = 10;
+
+	private lastEventTime = 0;
+	private storageEventCount = 0;
+	private async = false;
+
+	constructor(defaultValue: T) {
+		super(defaultValue);
+	}
+
+	public override set value(value: T) {
+		const actualSet = () => {
+			super.value = value;
+
+			const now = Date.now();
+			const timeDiff = now - this.lastEventTime;
+
+			this.storageEventCount++;
+
+			if (timeDiff < RateLimitedObservable.timeWindow) {
+				if (this.storageEventCount > RateLimitedObservable.maxEvents) {
+					this.async = true;
+					console.error('Too many observable events in a short time window!!!', new Error().stack, this);
+				}
+			} else {
+				this.storageEventCount = 0;
```

Comment:
Tiny, unimportant nitpick :smile: :
```suggestion
				this.storageEventCount = 1;
```